In [5]:
import pandas as pd
#!pip install sidetable
import sidetable

- First I'll be trying to answer one of the questions

Namenly:Which districts/neighborhoods have the most accidents?

-Whichone of the datasets do I need to work with? Is it one or more than one?
So far i think only the Accidents one.

In [6]:
path = r'C:\Users\Usuario\Desktop\gits\Project-Week-2-Barcelona\datasets'
dataset_acc = pd.read_csv(path+'/'+'2019_accidents_gu_bcn.csv')
dataset_acc

Numero_expedient  Codi_districte        Nom_districte  Codi_barri  \
0      2019S000001                   5  Sarrià-Sant Gervasi          26   
1      2019S000002                   9          Sant Andreu          62   
2      2019S000003                  10           Sant Martí          66   
3      2019S000004                   6               Gràcia          32   
4      2019S000005                   2             Eixample           7   
...                ...             ...                  ...         ...   
10022  2019S010046                   4            Les Corts          21   
10023  2019S010047                  10           Sant Martí          64   
10024  2019S010048                   2             Eixample           5   
10025  2019S010049                   4            Les Corts          19   
10026  2019S010050                   2             Eixample           6   

                                Nom_barri  Codi_carrer  \
0                  Sant Gervasi - Galvany       144601   
1                el Congrés i els Indians       102907   
2       el Parc i la Llacuna del Poblenou       242906   
3      el Camp d'en Grassot i Gràcia Nova       228803   
4                  la Dreta de l'Eixample        89004   
...                                   ...          ...   
10022                           Pedralbes       101700   
10023          el Camp de l'Arpa del Clot       297001   
10024                       el Fort Pienc        28305   
10025                           les Corts       167800   
10026                  la Sagrada Família       178308   

                                              Nom_carrer Num_postal_caption  \
0      Diagonal / Augusta                            ...          0482 0482   
1      Felip II / Congrés Eucarístic                 ...          9999 9999   
2      Pallars                                       ...          0111 0113   
3      Taxdirt / Nogués                              ...          0054 0054   
4      Consell de Cent / Girona                      ...          0395 0397   
...                                                  ...                ...   
10022  Doctor Ferran / Manila                        ...          0025 0025   
10023  Sant Antoni Maria Claret                      ...          0346 0348   
10024  Ausiàs Marc / Nàpols                          ...          0080 0080   
10025  Joaquim Molins                                ...          0009 0009   
10026  Lepant                                        ...          0255 0255   

      Descripcio_dia_setmana Dia_setmana  ...  \
0                    Dimarts          Dm  ...   
1                    Dimarts          Dm  ...   
2                    Dimarts          Dm  ...   
3                    Dimarts          Dm  ...   
4                    Dimarts          Dm  ...   
...                      ...         ...  ...   
10022                Dimarts          Dm  ...   
10023                Dimarts          Dm  ...   
10024                Dimarts          Dm  ...   
10025                Dimarts          Dm  ...   
10026               Dimecres          Dc  ...   

              Descripcio_causa_vianant  Numero_morts  Numero_lesionats_lleus  \
0             No és causa del  vianant             0                       0   
1             No és causa del  vianant             0                       1   
2      Creuar per fora pas de vianants             0                       1   
3             No és causa del  vianant             0                       1   
4             No és causa del  vianant             0                       2   
...                                ...           ...                     ...   
10022         No és causa del  vianant             0                       1   
10023         No és causa del  vianant             0                       1   
10024         No és causa del  vianant             0                       1   
10025         No és causa del  vianant             0                  

Let's see what is the total of accidents in Barcelona in 2019

In [7]:
num_accidents = dataset_acc.shape[0]
num_accidents

10027

We need to count the number of ocurrences of each district, this will be easy because we have the variable both encoded and decoded (string).

In [26]:
acc_district = dataset_acc.stb.freq(['Nom_districte'])

print(acc_district)

acc_district[1,1]

          Nom_districte  Count   Percent  Cumulative Count  Cumulative Percent
0              Eixample   3010  0.300189              3010            0.300189
1            Sant Martí   1243  0.123965              4253            0.424155
2        Sants-Montjuïc   1172  0.116884              5425            0.541039
3   Sarrià-Sant Gervasi   1053  0.105016              6478            0.646056
4             Les Corts    668  0.066620              7146            0.712676
5        Horta-Guinardó    664  0.066221              7810            0.778897
6           Sant Andreu    643  0.064127              8453            0.843024
7          Ciutat Vella    564  0.056248              9017            0.899272
8            Nou Barris    528  0.052658              9545            0.951930
9                Gràcia    463  0.046175             10008            0.998105
10           Desconegut     19  0.001895             10027            1.000000


KeyError: (1, 1)

In [9]:
acc_nhood = dataset_acc.stb.freq(['Nom_barri'])
acc_nhood.head(50)

Nom_barri  Count   Percent  \
0                         la Dreta de l'Eixample   1174  0.117084   
1                l'Antiga Esquerra de l'Eixample    565  0.056348   
2                         Sant Gervasi - Galvany    417  0.041588   
3                 la Nova Esquerra de l'Eixample    398  0.039693   
4                             la Sagrada Família    368  0.036701   
5                                  el Fort Pienc    302  0.030119   
6                                      les Corts    288  0.028722   
7                     la Marina del Prat Vermell    279  0.027825   
8                                   el Poble-sec    232  0.023138   
9                                    Sant Antoni    203  0.020245   
10                    Sant Gervasi - la Bonanova    201  0.020046   
11                    la Maternitat i Sant Ramon    200  0.019946   
12                                     Pedralbes    180  0.017952   
13                                   el Poblenou    178  0.017752   
14                                         Sants    178  0.017752   
15                       Provençals del Poblenou    171  0.017054   
16                                   Sant Andreu    169  0.016854   
17                                       el Clot    168  0.016755   
18                                        Sarrià    166  0.016555   
19                                 el Bon Pastor    154  0.015359   
20                                      el Raval    153  0.015259   
21                                la Barceloneta    150  0.014960   
22                                   Hostafrancs    150  0.014960   
23             el Parc i la Llacuna del Poblenou    139  0.013863   
24                    el Camp de l'Arpa del Clot    136  0.013563   
25                                el Barri Gòtic    134  0.013364   
26                               les Tres Torres    130  0.012965   
27            el Camp d'en Grassot i Gràcia Nova    129  0.012865   
28         Sant Pere, Santa Caterina i la Ribera    127  0.012666   
29                             la Vila de Gràcia    121  0.012067   
30                                         Horta    120  0.011968   
31  Diagonal Mar i el Front Marítim del Poblenou    119  0.011868   
32                          el Putxet i el Farró    118  0.011768   
33                              el Baix Guinardó    116  0.011569   
34                 la Vila Olímpica del Poblenou    113  0.011270   
35                                    la Bordeta    111  0.011070   
36                     Vallcarca i els Penitents    109  0.010871   
37                             la Marina de Port     99  0.009873   
38                                   el Guinardó     96  0.009574   
39                                la Prosperitat     88  0.008776   
40                                    la Sagrera     87  0.008677   
41                                     el Carmel     86  0.008577   
42                                         Navas     80  0.007978   
43                      Sant Martí de Provençals     79  0.007879   
44                             la Trinitat Vella     78  0.007779   
45                                      la Salut     77  0.007679   
46                                         Porta     77  0.007679   
47                           la Verneda i la Pau     74  0.007380   
48                 Vilapicina i la Torre Llobeta     73  0.007280   
49                              la Vall d'Hebron     67  0.006682   

    Cumulative Count  Cumulative Percent  
0               1174            0.117084  
1               1739            0.173432  
2               2156            0.215019  
3               2554            0.254712  
4               2922            0.291413  
5               3224            0.321532  
6               3512            0.350254  
7               3791            0.378079  
8               4023            0.401217  
9               4226            0.421462  
10              4427            0.441508  
11 

In [10]:
#acc_nest = dataset_acc.groupby(['Nom_districte','Nom_barri']).describe()
#acc_nest

Now Im gonna import the table that I found in this source https://www.bcn.cat/estadistica/castella/dades/inf/lecpadro/lec19/t12.htm
which is population by district to gain more insights

In [11]:
dataset_pop_dist = pd.read_csv(path+'/'+'poblacio_districte.csv')
dataset_pop_dist

Distritos  Unnamed: 1      TOTAL    %  Hombres   %.1  \
0                BARCELONA         NaN  1.650.358  100  782.555  47,4   
1          1. Ciutat Vella         NaN    105.820  100   56.107    53   
2              2. Eixample         NaN    269.095  100  125.612  46,7   
3        3. Sants-Montjuïc         NaN    185.450  100   88.853  47,9   
4             4. Les Corts         NaN     82.591  100   38.367  46,5   
5   5. Sarrià-Sant Gervasi         NaN    150.446  100   69.045  45,9   
6                6. Gràcia         NaN    122.853  100   56.404  45,9   
7        7. Horta-Guinardó         NaN    172.473  100   80.937  46,9   
8            8. Nou Barris         NaN    171.290  100   80.371  46,9   
9           9. Sant Andreu         NaN    150.264  100   71.335  47,5   
10          10. Sant Martí         NaN    240.076  100  115.524  48,1   

    Mujeres   %.2  
0   867.803  52,6  
1    49.713    47  
2   143.483  53,3  
3    96.597  52,1  
4    44.224  53,5  
5    81.401  54,1  
6    66.449  54,1  
7    91.536  53,1  
8    90.919  53,1  
9    78.929  52,5  
10  124.552  51,9

This one we do need to clean, but will be worth it because we will use it to answer the following questions too.

In [12]:
dataset_pop_dist1 = dataset_pop_dist.copy()
data_dist2 = dataset_pop_dist1.drop(['Unnamed: 1'], axis=1)
data_dist2

Distritos      TOTAL    %  Hombres   %.1  Mujeres   %.2
0                BARCELONA  1.650.358  100  782.555  47,4  867.803  52,6
1          1. Ciutat Vella    105.820  100   56.107    53   49.713    47
2              2. Eixample    269.095  100  125.612  46,7  143.483  53,3
3        3. Sants-Montjuïc    185.450  100   88.853  47,9   96.597  52,1
4             4. Les Corts     82.591  100   38.367  46,5   44.224  53,5
5   5. Sarrià-Sant Gervasi    150.446  100   69.045  45,9   81.401  54,1
6                6. Gràcia    122.853  100   56.404  45,9   66.449  54,1
7        7. Horta-Guinardó    172.473  100   80.937  46,9   91.536  53,1
8            8. Nou Barris    171.290  100   80.371  46,9   90.919  53,1
9           9. Sant Andreu    150.264  100   71.335  47,5   78.929  52,5
10          10. Sant Martí    240.076  100  115.524  48,1  124.552  51,9

We also drop the % column

In [13]:
data_dist3 = data_dist2.drop(['%'], axis=1)
data_dist3

Distritos      TOTAL  Hombres   %.1  Mujeres   %.2
0                BARCELONA  1.650.358  782.555  47,4  867.803  52,6
1          1. Ciutat Vella    105.820   56.107    53   49.713    47
2              2. Eixample    269.095  125.612  46,7  143.483  53,3
3        3. Sants-Montjuïc    185.450   88.853  47,9   96.597  52,1
4             4. Les Corts     82.591   38.367  46,5   44.224  53,5
5   5. Sarrià-Sant Gervasi    150.446   69.045  45,9   81.401  54,1
6                6. Gràcia    122.853   56.404  45,9   66.449  54,1
7        7. Horta-Guinardó    172.473   80.937  46,9   91.536  53,1
8            8. Nou Barris    171.290   80.371  46,9   90.919  53,1
9           9. Sant Andreu    150.264   71.335  47,5   78.929  52,5
10          10. Sant Martí    240.076  115.524  48,1  124.552  51,9

We are now going to clean the names of the districts so they match with the ones in the other dataframe

In [14]:
#Defining the function to clean the string
def cleaner_str(x):
    for index,i in enumerate(x):
        if x != 'BARCELONA':
            new_i = x[3:]
        
        else:
            new_i = x
    return new_i

#Applying the variable to each row in the column Distritos
data_dist3['Distritos'] = data_dist3['Distritos'].apply(cleaner_str)
data_dist3

Distritos      TOTAL  Hombres   %.1  Mujeres   %.2
0             BARCELONA  1.650.358  782.555  47,4  867.803  52,6
1          Ciutat Vella    105.820   56.107    53   49.713    47
2              Eixample    269.095  125.612  46,7  143.483  53,3
3        Sants-Montjuïc    185.450   88.853  47,9   96.597  52,1
4             Les Corts     82.591   38.367  46,5   44.224  53,5
5   Sarrià-Sant Gervasi    150.446   69.045  45,9   81.401  54,1
6                Gràcia    122.853   56.404  45,9   66.449  54,1
7        Horta-Guinardó    172.473   80.937  46,9   91.536  53,1
8            Nou Barris    171.290   80.371  46,9   90.919  53,1
9           Sant Andreu    150.264   71.335  47,5   78.929  52,5
10           Sant Martí    240.076  115.524  48,1  124.552  51,9

In [15]:
print(acc_district['Nom_districte'],data_dist3['Distritos'])


0                Eixample
1              Sant Martí
2          Sants-Montjuïc
3     Sarrià-Sant Gervasi
4               Les Corts
5          Horta-Guinardó
6             Sant Andreu
7            Ciutat Vella
8              Nou Barris
9                  Gràcia
10             Desconegut
Name: Nom_districte, dtype: object 0               BARCELONA
1            Ciutat Vella
2                Eixample
3          Sants-Montjuïc
4               Les Corts
5     Sarrià-Sant Gervasi
6                  Gràcia
7          Horta-Guinardó
8              Nou Barris
9             Sant Andreu
10             Sant Martí
Name: Distritos, dtype: object


Before merging we need the two columns to be called the same

In [16]:
#We rename the column to do the merge on it and also replace the nme Sant Marti because I tried the merge once and it didnt catch it
#witch means it has something different.Its the order!
acc_district.rename(columns={'Nom_districte':'Distritos'}, inplace=True)
#acc_district = acc_district.replace(to_replace ="Sant Martí", value ="Sant Martí")
acc_district.index(['Ciutat Vella','Eixample','Sants-Montjuïc','Les Corts','Sarrià-Sant Gervasi','Gràcia','Horta-Guinardó','Nou Barris','Sant Andreu','Sant Martí'])

TypeError: 'Int64Index' object is not callable

In [ ]:
#But we lost Sant Marti  on the first try of the merge so we have to fix it
# it is finally not worth it because we have no population data for it.

In [18]:
#This is the merge of the population table with the cumulative table from accidents but we gotta do it for the whole table accidents too.
merged_district =acc_district.merge(right=data_dist3, how='inner', on='Distritos')
merged_district

Distritos  Count   Percent  Cumulative Count  Cumulative Percent  \
0             Eixample   3010  0.300189              3010            0.300189   
1       Sants-Montjuïc   1172  0.116884              5425            0.541039   
2  Sarrià-Sant Gervasi   1053  0.105016              6478            0.646056   
3            Les Corts    668  0.066620              7146            0.712676   
4       Horta-Guinardó    664  0.066221              7810            0.778897   
5          Sant Andreu    643  0.064127              8453            0.843024   
6         Ciutat Vella    564  0.056248              9017            0.899272   
7           Nou Barris    528  0.052658              9545            0.951930   
8               Gràcia    463  0.046175             10008            0.998105   

     TOTAL  Hombres   %.1  Mujeres   %.2  
0  269.095  125.612  46,7  143.483  53,3  
1  185.450   88.853  47,9   96.597  52,1  
2  150.446   69.045  45,9   81.401  54,1  
3   82.591   38.367  46,5   44.224  53,5  
4  172.473   80.937  46,9   91.536  53,1  
5  150.264   71.335  47,5   78.929  52,5  
6  105.820   56.107    53   49.713    47  
7  171.290   80.371  46,9   90.919  53,1  
8  122.853   56.404  45,9   66.449  54,1

In [101]:
#We change some column names to make it more readable.
merged_district = merged_district.rename(columns={'Percent':'Percent_acci','Cumulative Count':'Cumulative_Count','Cumulative Percent':'Cumulative_Percent','TOTAL':'Total_Pop'})

In [105]:
merged_district

Distritos  Count   Percent  Cumulative Count  Cumulative Percent  \
0             Eixample   3010  0.300189              3010            0.300189   
1       Sants-Montjuïc   1172  0.116884              5425            0.541039   
2  Sarrià-Sant Gervasi   1053  0.105016              6478            0.646056   
3            Les Corts    668  0.066620              7146            0.712676   
4       Horta-Guinardó    664  0.066221              7810            0.778897   
5          Sant Andreu    643  0.064127              8453            0.843024   
6         Ciutat Vella    564  0.056248              9017            0.899272   
7           Nou Barris    528  0.052658              9545            0.951930   
8               Gràcia    463  0.046175             10008            0.998105   

     TOTAL  Hombres   %.1  Mujeres   %.2  
0  269.095  125.612  46,7  143.483  53,3  
1  185.450   88.853  47,9   96.597  52,1  
2  150.446   69.045  45,9   81.401  54,1  
3   82.591   38.367  46,5   44.224  53,5  
4  172.473   80.937  46,9   91.536  53,1  
5  150.264   71.335  47,5   78.929  52,5  
6  105.820   56.107    53   49.713    47  
7  171.290   80.371  46,9   90.919  53,1  
8  122.853   56.404  45,9   66.449  54,1

In [56]:
#We have to replace the ',' in %.1 and %.2 by '.' to convert into float
merged_district['%.1'] = merged_district['%.1'].apply(lambda x: x.replace(',', '.')).astype('float')
merged_district['%.2'] = merged_district['%.2'].apply(lambda x: x.replace(',', '.')).astype('float')
merged_district

Distritos  Count  Percent_acci  Cumulative_Count  \
0             Eixample   3010      0.300189              3010   
1       Sants-Montjuïc   1172      0.116884              5425   
2  Sarrià-Sant Gervasi   1053      0.105016              6478   
3            Les Corts    668      0.066620              7146   
4       Horta-Guinardó    664      0.066221              7810   
5          Sant Andreu    643      0.064127              8453   
6         Ciutat Vella    564      0.056248              9017   
7           Nou Barris    528      0.052658              9545   
8               Gràcia    463      0.046175             10008   

   Cumulative_Percent Total_Pop  Hombres   %.1  Mujeres   %.2  
0            0.300189   269.095  125.612  46.7  143.483  53.3  
1            0.541039   185.450   88.853  47.9   96.597  52.1  
2            0.646056   150.446   69.045  45.9   81.401  54.1  
3            0.712676    82.591   38.367  46.5   44.224  53.5  
4            0.778897   172.473   80.937  46.9   91.536  53.1  
5            0.843024   150.264   71.335  47.5   78.929  52.5  
6            0.899272   105.820   56.107  53.0   49.713  47.0  
7            0.951930   171.290   80.371  46.9   90.919  53.1  
8            0.998105   122.853   56.404  45.9   66.449  54.1

In [57]:
#We make sure all numeric columns are set to int or float.
cols =['Percent_acci','Cumulative_Percent','Total_Pop','Hombres','Mujeres']
merged_district[cols] = merged_district[cols].astype(float)

In [60]:
#We create a column Pop_Percent to know which percentatge of the population represents each district 
merged_district['Pop_Percent'] = ((merged_district['Total_Pop'] / merged_district['Total_Pop'].sum())*100).round(2)
merged_district

Distritos  Count  Percent_acci  Cumulative_Count  \
0             Eixample   3010      0.300189              3010   
1       Sants-Montjuïc   1172      0.116884              5425   
2  Sarrià-Sant Gervasi   1053      0.105016              6478   
3            Les Corts    668      0.066620              7146   
4       Horta-Guinardó    664      0.066221              7810   
5          Sant Andreu    643      0.064127              8453   
6         Ciutat Vella    564      0.056248              9017   
7           Nou Barris    528      0.052658              9545   
8               Gràcia    463      0.046175             10008   

   Cumulative_Percent  Total_Pop  Hombres   %.1  Mujeres   %.2  Pop_Percent  
0            0.300189    269.095  125.612  46.7  143.483  53.3        19.08  
1            0.541039    185.450   88.853  47.9   96.597  52.1        13.15  
2            0.646056    150.446   69.045  45.9   81.401  54.1        10.67  
3            0.712676     82.591   38.367  46.5   44.224  53.5         5.86  
4            0.778897    172.473   80.937  46.9   91.536  53.1        12.23  
5            0.843024    150.264   71.335  47.5   78.929  52.5        10.65  
6            0.899272    105.820   56.107  53.0   49.713  47.0         7.50  
7            0.951930    171.290   80.371  46.9   90.919  53.1        12.15  
8            0.998105    122.853   56.404  45.9   66.449  54.1         8.71

In [62]:
#We compute the % of population this 3 districts accumulate
merged_district.iloc[:3, [10]].sum()

Pop_Percent    42.9
dtype: float64

In [63]:
merged_district.iloc[:3, [7]].sum()

%.1    140.5
dtype: float64

In [64]:
merged_district.corr()

Count  Percent_acci  Cumulative_Count  \
Count               1.000000      1.000000         -0.898320   
Percent_acci        1.000000      1.000000         -0.898320   
Cumulative_Count   -0.898320     -0.898320          1.000000   
Cumulative_Percent -0.898320     -0.898320          1.000000   
Total_Pop           0.828321      0.828321         -0.695553   
Hombres             0.825302      0.825302         -0.690564   
%.1                -0.169049     -0.169049          0.198054   
Mujeres             0.825090      0.825090         -0.694910   
%.2                 0.169049      0.169049         -0.198054   
Pop_Percent         0.828310      0.828310         -0.695640   

                    Cumulative_Percent  Total_Pop   Hombres       %.1  \
Count                        -0.898320   0.828321  0.825302 -0.169049   
Percent_acci                 -0.898320   0.828321  0.825302 -0.169049   
Cumulative_Count              1.000000  -0.695553 -0.690564  0.198054   
Cumulative_Percent            1.000000  -0.695553 -0.690564  0.198054   
Total_Pop                    -0.695553   1.000000  0.995517 -0.253944   
Hombres                      -0.690564   0.995517  1.000000 -0.162400   
%.1                           0.198054  -0.253944 -0.162400  1.000000   
Mujeres                      -0.694910   0.996807  0.984786 -0.329452   
%.2                          -0.198054   0.253944  0.162400 -1.000000   
Pop_Percent                  -0.695640   1.000000  0.995477 -0.254348   

                     Mujeres       %.2  Pop_Percent  
Count               0.825090  0.169049     0.828310  
Percent_acci        0.825090  0.169049     0.828310  
Cumulative_Count   -0.694910 -0.198054    -0.695640  
Cumulative_Percent -0.694910 -0.198054    -0.695640  
Total_Pop           0.996807  0.253944     1.000000  
Hombres             0.984786  0.162400     0.995477  
%.1                -0.329452 -1.000000    -0.254348  
Mujeres             1.000000  0.329452     0.996840  
%.2                 0.329452  1.000000     0.254348  
Pop_Percent         0.996840  0.254348     1.000000

Insights:  We see thatas expected population is positively correlated with the number of accidents and that this correclation is fairly equal for men and women.64,6% of the accidents happen in just 3 districts (Eixample, Sants and Sarrià) which have 42.9% of population.We could gain more insights with more data so now we will proceed to answer the following question.

- Which districts have the more victims?

In [88]:
# we first load the next data set
dataset_vict = pd.read_csv(path+'/'+'2019_accidents_persones_gu_bcn_.csv')

In [67]:
#now we are gonna check it and get rid of unecesary cloumns
dataset_vict

Numero_Expedient  Codi_districte        Nom_districte  Codi_barri  \
0      2019S003731                   5  Sarrià-Sant Gervasi          22   
1      2019S007079                   5  Sarrià-Sant Gervasi          22   
2      2019S000173                   5  Sarrià-Sant Gervasi          22   
3      2019S001526                   4            Les Corts          21   
4      2019S005025                   4            Les Corts          21   
...                ...             ...                  ...         ...   
11839  2019S010041                   5  Sarrià-Sant Gervasi          23   
11840  2019S010041                   5  Sarrià-Sant Gervasi          23   
11841  2019S010044                   2             Eixample           6   
11842  2019S010044                   2             Eixample           6   
11843  2019S010050                   2             Eixample           6   

                                   Nom_barri  Codi_carrer  \
0      Vallvidrera, el Tibidabo i les Planes       700059   
1      Vallvidrera, el Tibidabo i les Planes       700059   
2      Vallvidrera, el Tibidabo i les Planes       107853   
3                                  Pedralbes       701267   
4                                  Pedralbes       701267   
...                                      ...          ...   
11839                                 Sarrià       701267   
11840                                 Sarrià       701267   
11841                     la Sagrada Família        90502   
11842                     la Sagrada Família        90502   
11843                     la Sagrada Família       178308   

                                              Num_postal Num_postal.1  \
0      Vallvidrera a les Planes / Vesc               ...    0050 0050   
1      Vallvidrera a les Planes                      ...       520000   
2      Robínia                                       ...    0001 0001   
3      B-20 (Besòs)                                  ...       460000   
4      Dalt (Besòs)                                  ...       480000   
...                                                  ...          ...   
11839                                                NaN            9   
11840                                                NaN            9   
11841                                                NaN    0597 0597   
11842                                                NaN    0597 0597   
11843                                                NaN    0255 0255   

      Descripcio_dia_setmana Dia_setmana  ... Edat  Descripcio_tipus_persona  \
0                   Dimecres          Dc  ...   30                 Conductor   
1                  Divendres          Dv  ...   47                 Conductor   
2                    Dimarts          Dm  ...   55                 Conductor   
3                    Dimarts          Dm  ...   40                 Conductor   
4                     Dijous          Dj  ...   30                 Conductor   
...                      ...         ...  ...  ...                       ...   
11839                Dimarts          Dm  ...   28                 Passatger   
11840                Dimarts          Dm  ...   24                 Conductor   
11841                Dimarts          Dm  ...   24                   Vianant   
11842                Dimarts          Dm  ...   24                   Vianant   
11843               Dimecres          Dc  ...   53                 Conductor   

                    Descripcio_Lloc_atropellament_vianat  \
0      N/A                                           ...   
1      N/A                                           ...   
2      N/A                                           ...   
3      N/A                                           ...   
4      N/A                                           ...   
...                                                  ...   
11839  N/A                                           ...   
11840  N/A                                           .

In [89]:
dataset_vict.columns
dataset_vict = dataset_vict.drop(['Numero_Expedient', 'Codi_barri',
       'Nom_barri', 'Codi_carrer', 'Num_postal', 'Num_postal.1',
       'Descripcio_dia_setmana', 'Dia_setmana', 'Descripcio_tipus_dia',
       'NK_ Any', 'Mes_ any', 'Nom_mes', 'Dia_mes', 'Descripcio_torn',
       'Hora_dia','Coordenada_UTM_X', 'Coordenada_UTM_Y', 'Longitud', 'Latitud'], axis = 1)

In [90]:
# We are going to also drop Descripcio_Lloc_atropellament_viananant
dataset_vict = dataset_vict.drop(['Descripcio_Lloc_atropellament_vianat'],axis = 1)

In [92]:
#Also cleaned the categorical value a bit
dataset_vict = dataset_vict.replace(to_replace = "Es desconeix", value ="Desconegut")

In [93]:
dataset_vict

Codi_districte        Nom_districte         Descripcio_causa_vianant  \
0                   5  Sarrià-Sant Gervasi         No és causa del  vianant   
1                   5  Sarrià-Sant Gervasi         No és causa del  vianant   
2                   5  Sarrià-Sant Gervasi         No és causa del  vianant   
3                   4            Les Corts         No és causa del  vianant   
4                   4            Les Corts         No és causa del  vianant   
...               ...                  ...                              ...   
11839               5  Sarrià-Sant Gervasi         No és causa del  vianant   
11840               5  Sarrià-Sant Gervasi         No és causa del  vianant   
11841               2             Eixample  Creuar per fora pas de vianants   
11842               2             Eixample  Creuar per fora pas de vianants   
11843               2             Eixample         No és causa del  vianant   

      Desc_Tipus_vehicle_implicat Descripcio_sexe Edat  \
0                     Motocicleta            Home   30   
1                       Furgoneta            Home   47   
2                         Turisme            Dona   55   
3                     Motocicleta            Dona   40   
4                     Motocicleta            Home   30   
...                           ...             ...  ...   
11839                 Motocicleta            Dona   28   
11840                 Motocicleta            Home   24   
11841                 Motocicleta            Dona   24   
11842                 Motocicleta            Home   24   
11843                 Motocicleta            Home   53   

      Descripcio_tipus_persona Descripcio_Motiu_desplaçament_vianant  \
0                    Conductor                            Desconegut   
1                    Conductor                            Desconegut   
2                    Conductor                            Desconegut   
3                    Conductor                            Desconegut   
4                    Conductor                            Desconegut   
...                        ...                                   ...   
11839                Passatger                            Desconegut   
11840                Conductor                            Desconegut   
11841                  Vianant                   Oci i entreteniment   
11842                  Vianant                   Oci i entreteniment   
11843                Conductor                            Desconegut   

      Descripcio_Motiu_desplaçament_conductor  \
0                                  In itinere   
1       Transport professional de mercaderies   
2                                  Desconegut   
3                                  In itinere   
4                                  Desconegut   
...                                       ...   
11839                              Desconegut   
11840                     Oci i entreteniment   
11841                              Desconegut   
11842                              Desconegut   
11843                              In itinere   

                                Descripcio_victimitzacio  
0             Ferit greu: hospitalització superior a 24h  
1              Ferit lleu: Rebutja assistència sanitària  
2                 Ferit lleu: Hospitalització fins a 24h  
3                 Ferit lleu: Hospitalització fins a 24h  
4      Ferit lleu: Amb assistència sanitària en lloc ...  
...                                                  ...  
11839             Ferit lleu: Hospitalització fins a 24h  
11840  Ferit lleu: Amb assistència sanitària en lloc ...  
11841         Ferit greu: hospitalització superior a 24h  
11842         Ferit greu: hospitalització superior a 24h  
11843         Ferit greu: hospitalització superior a 24h  

[11844 rows x 10 columns]

In [95]:
acc_district_vict = dataset_vict.stb.freq(['Nom_districte'])
acc_district_vict

Nom_districte  Count   Percent  Cumulative Count  Cumulative Percent
0              Eixample   3603  0.304205              3603            0.304205
1            Sant Martí   1503  0.126900              5106            0.431104
2        Sants-Montjuïc   1377  0.116261              6483            0.547366
3   Sarrià-Sant Gervasi   1253  0.105792              7736            0.653158
4             Les Corts    832  0.070247              8568            0.723404
5        Horta-Guinardó    784  0.066194              9352            0.789598
6           Sant Andreu    781  0.065941             10133            0.855539
7          Ciutat Vella    585  0.049392             10718            0.904931
8            Nou Barris    582  0.049139             11300            0.954070
9                Gràcia    525  0.044326             11825            0.998396
10           Desconegut     19  0.001604             11844            1.000000